src/aics-image-viewer/components/ControlPanel/index.tsx

```diff
@@ -41,6 +41,9 @@ const ControlTabNames = {
 export default function ControlPanel(props: ControlPanelProps): React.ReactElement {
   const [tab, setTab] = React.useState(ControlTab.Channels);
 
+  const controlPanelContainerRef = React.useRef<HTMLDivElement>(null);
+  const getDropdownContainer = controlPanelContainerRef.current ? () => controlPanelContainerRef.current! : undefined;
+
```

Comment:
This pattern is used in a few places, but it prevents Ant from directly appending menus and popups to the `body` tag. Instead, they'll be anchored in the HTML tree to wherever the `containerRef` is, allowing them to receive style overrides from `StyleProvider`.

Reply:
This solution is low-impact enough to be acceptable, but thinking about doing this every time we need a dropdown does make me pause and wonder if there's a nicer place to hold our global styles long-term. Should the CSS applied to `StyleProvider` live in a class that we can name and apply to Ant's container via `overlayClassName`? If so, would we be able to build the theme in from JavaScript as easily as `StyleProvider` does now (maybe with another build step)? Would it be as easy as it is now to maintain a single source of truth for theme colors (probably not)? Food for thought as we think about how to style this app going forward.

Reply:
In colorizer this is handled by `AccessibleDropdown`! Making a component that does the container management is a good way to handle it.

Using `overlayClassName` would require us to attach CSS rules to the body or use `:root`, both of which could potentially lead to conflicts with other styles when we embed the viewer... But I see what you're saying and yes, let's keep thinking about style management.